libimage/pull.go

```diff
@@ -379,6 +381,48 @@ func (r *Runtime) copyFromRegistry(ctx context.Context, ref types.ImageReference
 		return nil, err
 	}
 
+	// Avoid running out of time when running inside a systemd unit by
+	// regularly increasing the timeout.
+	if socketPath, ok := os.LookupEnv("NOTIFY_SOCKET"); ok {
+		socketAddr := &net.UnixAddr{
+			Name: socketPath,
+			Net:  "unixgram",
+		}
+		conn, err := net.DialUnix(socketAddr.Net, nil, socketAddr)
+		if err != nil {
+			return nil, err
+		}
+		defer conn.Close()
+
+		timeout := 30 * time.Second
+		timer := time.NewTimer(timeout)
+		socketCtx, cancel := context.WithCancel(ctx)
+		defer cancel()
+		defer timer.Stop()
+
+		// See upstream systemd change adding support for EXTEND_TIMEOUT_USEC:
+		// https://github.com/systemd/systemd/commit/a327431bd168b2f327f3cd422379e213c643f2a5
+		extendValue := []byte(fmt.Sprintf("EXTEND_TIMEOUT_USEC=%d", timeout.Microseconds()))
+		extendTimeout := func() {
+			if _, err := conn.Write(extendValue); err != nil {
```

Comment:
Can we put a loginfo.Info or Warning to say that the timeout was extended, That way the user will know that Podman is extending the timeout for pull.